### PR TITLE
ci/gpu: fix install future & use local cache

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

use local cache of persistent hosted runners

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

cc @carmocca @akihironitta @borda

### CODE REVIEW

**Review by ethanwharris — APPROVED** [review]

**Comment by Borda** [comment]
It seems PT test storage is broken:
```
Looking in links: https://download.pytorch.org/whl/test/cu117/torch_test.html
Collecting torch>=2.0.0
  Downloading https://download.pytorch.org/whl/test/cu117/torch-2.0.0%2Bcu117-cp39-cp39-linux_x86_64.whl (1843.9 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.8/1.8 GB 22.4 MB/s eta 0:00:00
Collecting torchvision>=0.15.0
  Downloading https://download.pytorch.org/whl/test/cu117/torchvision-0.15.0%2Bcu117-cp39-cp39-linux_x86_64.whl (6.2 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 6.2/6.2 MB 23.4 MB/s eta 0:00:00
Requirement already satisfied: sympy in /usr/local/lib/python3.9/dist-packages (from torch>=2.0.0) (1.11.1)
Requirement already satisfied: typing-extensions in /usr/local/lib/python3.9/dist-packages (from torch>=2.0.0) (4.4.0)
Requirement already satisfied: filelock in /usr/local/lib/python3.9/dist-packages (from torch>=2.0.0) (3.9.0)
ERROR: Could not find a version that satisfies the requirement pytorch-triton==2.0.0+b8b470bc59 (from torch) (from versions: 0.0.1, 2.0.0+0d7e753227, 2.0.0+c8bfe3f548, 2.0.0+d54c04abe2)
ERROR: No matching distribution found for pytorch-triton==2.0.0+b8b470bc59
```

**Comment by Borda** [comment]
turned out that they mixed packages/dependencies from test and nightly pages :otter:

File: .github/workflows/ci-dockers-pytorch.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ on:
     types: [opened, reopened, ready_for_review, synchronize]  # added `ready_for_review` since draft is skipped
     paths:
       - ".actions/**"
-      - ".github/workflows/ci-pytorch-dockers.yml"
+      - ".github/workflows/ci-dockers-pytorch.yml"
```
Comment (carmocca) [inline .github/workflows/ci-dockers-pytorch.yml:11]: @Borda this change needs to be done to the checkgroup config too.

The paths and checkgroup config have to be the same

File: .github/workflows/ci-dockers-pytorch.yml [hunk 2]
```diff
@@ -105,6 +105,7 @@ jobs:
           - {python_version: "3.9", pytorch_version: "1.11", cuda_version: "11.3.1"}
           - {python_version: "3.9", pytorch_version: "1.12", cuda_version: "11.6.1"}
           - {python_version: "3.9", pytorch_version: "1.13", cuda_version: "11.7.1"}
+          - {python_version: "3.10", pytorch_version: "1.13", cuda_version: "11.7.1"}
```
Comment (carmocca) [inline .github/workflows/ci-dockers-pytorch.yml:108]: This also needs to be added to the checkgroup config

File: .github/workflows/ci-dockers-pytorch.yml (hunk 1 above)
Comment (Borda) [reply .github/workflows/ci-dockers-pytorch.yml:11]: sorry, fixing it in #16984

**Pull request merged**